os-server/src/main/java/app/openschool/common/security/JwtTokenProvider.java

```diff
@@ -64,9 +66,13 @@ private boolean isTokenExpired(JWTVerifier verifier, String token) {
     return expiration.before(new Date());
   }
 
-  public String getSubject(String token) {
-    JWTVerifier verifier = getJwtVerifier();
-    return verifier.verify(token).getSubject();
+  public String getSubject(String token) throws InvalidTokenException {
+    try {
+      JWTVerifier verifier = getJwtVerifier();
+      return verifier.verify(token).getSubject();
+    } catch (SignatureVerificationException e) {
+      throw new InvalidTokenException(e.getMessage());
```

Comment:
please handle this exception in the exception handler and provide proper response body